utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -383,6 +399,268 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
+                                        end: c - 1,
+                                        value,
+                                    });
+                                }
+                                trie_value = trie_value_2; // may or may not help
+                            }
+                        } else {
+                            trie_value = trie_value_2;
+                            value = maybe_filter_value(
+                                trie_value_2,
+                                self.header.null_value,
+                                null_value,
+                            );
+                            have_value = true;
+                        }
+
+                        c += 1;
+                        while (c & data_mask) != 0 {
+                            di += 1;
+                            trie_value_2 = if let Some(trv2) = self.data.get(di as usize) {
+                                trv2.into()
+                            } else {
+                                return None;
+                            };
+                            if trie_value_2 != trie_value {
+                                if maybe_filter_value(
+                                    trie_value_2,
+                                    self.header.null_value,
+                                    null_value,
+                                ) != value
+                                {
+                                    return Some(CodePointMapRange {
+                                        start,
+                                        end: c - 1,
+                                        value,
+                                    });
+                                }
+                                trie_value = trie_value_2; // may or may not help
+                            }
+
+                            c += 1;
+                        }
+                    }
+                }
+
+                i3 += 1;
+                if i3 >= i3_block_length {
+                    break;
+                }
+            }
+
+            if c >= self.header.high_start {
+                break;
+            }
+        }
+
+        debug_assert!(have_value);
+
+        let di: u32 = self.data.len() as u32 - HIGH_VALUE_NEG_DATA_OFFSET;
+        let high_value: u32 = if let Some(hv) = self.data.get(di as usize) {
+            hv.into()
+        } else {
+            return None;
+        };
+        if maybe_filter_value(high_value, self.header.null_value, null_value) != value {
+            c -= 1;
+        } else {
+            c = CODE_POINT_MAX;
+        }
+        Some(CodePointMapRange {
+            start,
+            end: c,
+            value,
+        })
+    }
+
+    /// Yields an [`Iterator`] returning ranges of consecutive code points that
+    /// share the same value in the [`CodePointTrie`].
+    pub fn iter_ranges(&'trie self) -> CodePointMapRangeIterator<'trie, T> {
```

Comment:
Nit: Add an example to the docs.  Suggestion: Show all the ranges from the planes trie (there should be only 13 of them) by calling `.next()` 13 times until it returns `None`.